refactor DBManager class to multiple subclasses

Jhenderson19 opened the pull request:
also removed underscore.js

Jhenderson19 committed:
refactor DBManager class to multiple subclasses

also removed underscore.js

Jhenderson19 committed:
more code shuffling, preparing to implement postgress

anthonysim reviewed (approved):
I don't really have any suggestions. LGTM! Looks clean!

Jhenderson19 committed:
Update: Query Benchmarking

 - added log for attempting to use an unimplemented database
 - renamed benchmark.js in order to have multiple benchmarking scripts
 - added return null to cassandra's connection promise catch
 - cassandra's "Retry until success" strategy now only ignores
"BusyConnectionError" in order to avoid program hangs and catastrophic
failures in the event a more significant error is thrown
 - cassandra's getcourse only returns relevant data to the server, not
the whole response from the database
 - simplified cassandra's delete all courses query, made it more idiot
proof really
 - retry function that cassandra uses refactored to allow error
filtering
 - implemented postgres (somehow I thought I had already commited
this???)
 - added fancy loading bars to seed script :3
 - replaced calls to console.log with logger.log in seed script so that the output
is written to disk AND console.
 - cleaned up seed script log output, added more information
 - added perf_hooks to allow use of 'performance.now()' which has a
higher time resolution than Date.now()
 - added additional npm run comands for benchmarking
 - no longer keep reference to the DBManager class itself in the server
because we only need to instantiate one.

Jhenderson19 committed:
added some comments to benchmark query
